content/blog/2023/11/2023-11-29-jenkins-to-github.adoc

```diff
@@ -0,0 +1,269 @@
+When the builds are done, the results should be conveniently visible at the "Checks" section at the bottom of the PR, like this:
+
+image:/images/post-images/2023/11/29/20230902144232-Screenshot-of-a-typical-PR-Checks-list.png[Screenshot of a typical PR Checks list]
+
+== Ways to publish checks to PRs: Statuses, Checks, and Actions
+
+You thought about your options.
+There are 3 types of entries that can be shown in this little box.
+In increasing order of complexity in setting up --
+
+. *Statuses*, or as I'd like to call them, *Commit Checks*.
+They are associated with commits, not PRs.
+This means you can run them even without creating a PR.
+For example, here are some Commit Checks on a commit I directly pushed to a `master` branch:
+image:/images/post-images/2023/11/29/20230902150825-Screenshot-of-a-list-of-commit-statuses.png[Screenshot of a list of commit statuses]
+
+. *Checks*, or more explicitly, *PR Checks*.
+These are run on commits but are stored with the PRs.
+With them, you can write long essays of messages to the "Checks" tab of a PR, but they do require authenticating as a GitHub App instead of a user, so it's more complicated to set up.
+SonarCloud, a code analyzer, makes use of PR Checks via https://github.com/apps/sonarcloud[their namesake GitHub App].
+If you look closer, their report is just Markdown:
+image:/images/post-images/2023/11/29/20230902151934-Screenshot-of-a-SonarCloud-Code-Analysis-as-a-GitHub-PR-Check.png[Screenshot of a SonarCloud Code Analysis as a GitHub PR Check]
+
+. *GitHub Actions* (GA).
+GA is a different kind of CI system from Jenkins.
+Since this article is about Jenkins only, GA is only mentioned here for the sake of completeness.
+image:/images/post-images/2023/11/29/20230902152320-Screenshot-of-a-GitHub-Action-as-a-PR-Check.png[Screenshot of a GitHub Action as a PR Check]
+
+== When the result is simple: Use commit statuses
+
+The Jenkins jobs you want to trigger only runs for a couple of seconds.
+Each only emits a simple result: pass or fail.
+You don't need fancy reports, so *commit statuses* seem to be a suitable choice.
+
+You installed https://plugins.jenkins.io/github/[the GitHub plugin] to your Jenkins server and followed https://stackoverflow.com/a/51003334/1147061[this guide] to set it up.
+You are able to emit simple one-liners as checks now, like this:
+
+image:https://raw.githubusercontent.com/jenkinsci/github-coverage-reporter/readme/assets/coverage-success.png[]
+_(https://github.com/jenkinsci/github-coverage-reporter-plugin[source] of screenshot; Apache 2.0 licensed)_
+
+It met your immediate need, but it stills falls short in a number of ways:
+
+* The plugin publishes to PR only thrice: upon getting queued, upon the start, and upon completion.
+Your job actually has *many time-consuming stages* (e.g., building a manifest, deploying to staging, and running tests), often packed in a single script, and you *want to be notified at each stage*.
```

Comment:
```suggestion
Your job actually has *many* time-consuming stages, such as building a manifest, deploying to staging, and running tests. 
These stages are often packed in a single script and you *want to be notified at each stage*.
```
Suggestion to adjust sentence structure and add some syntax to convert to two sentences.